SignalServiceKit/src/Devices/OWSRecordTranscriptJob.m

```diff
@@ -79,24 +80,58 @@ - (void)runWithAttachmentHandler:(void (^)(TSAttachmentStream *attachmentStream)
         [[OWSAttachmentsProcessor alloc] initWithAttachmentProtos:transcript.attachmentPointerProtos
-                                                        timestamp:transcript.timestamp
                                                             relay:transcript.relay
-                                                           thread:thread
                                                    networkManager:self.networkManager
-                                                   primaryStorage:self.primaryStorage
                                                       transaction:transaction];
 
+
     // TODO group updates. Currently desktop doesn't support group updates, so not a problem yet.
     TSOutgoingMessage *outgoingMessage =
         [[TSOutgoingMessage alloc] initOutgoingMessageWithTimestamp:transcript.timestamp
-                                                           inThread:thread
+                                                           inThread:transcript.thread
                                                         messageBody:transcript.body
                                                       attachmentIds:[attachmentsProcessor.attachmentIds mutableCopy]
                                                    expiresInSeconds:transcript.expirationDuration
                                                     expireStartedAt:transcript.expirationStartedAt
                                                      isVoiceMessage:NO
                                                    groupMetaMessage:TSGroupMessageNone
-                                                      quotedMessage:nil];
+                                                      quotedMessage:transcript.quotedMessage];
+
+    // TODO: When written, desktop didn't yet support sending quotedMessages, so we didn't have a
+    // good way to test the handling of transcripts with a quotedMessage. This assertion can be delete
+    // once we've tested transcripts with quoted messages are processed correctly.
+    OWSAssert(transcript.quotedMessage == nil);
+
+    TSQuotedMessage *_Nullable quotedMessage = transcript.quotedMessage;
+    if (quotedMessage && quotedMessage.thumbnailAttachmentPointerId) {
+        // We weren't able to derive a local thumbnail, so we'll fetch the referenced attachment.
+        TSAttachmentPointer *attachmentPointer =
+            [TSAttachmentPointer fetchObjectWithUniqueID:quotedMessage.thumbnailAttachmentPointerId
+                                             transaction:transaction];
+
+        if ([attachmentPointer isKindOfClass:[TSAttachmentPointer class]]) {
+            OWSAttachmentsProcessor *attachmentProcessor =
+                [[OWSAttachmentsProcessor alloc] initWithAttachmentPointer:attachmentPointer
+                                                            networkManager:self.networkManager];
+
+            DDLogDebug(@"%@ downloading thumbnail for transcript: %tu", self.logTag, transcript.timestamp);
```

Comment:
Sorry, this PR is long and I may have missed it but: is this the only place where we initiate downloads of quoted reply thumbnails?  I suspect we'll want to initiate them in the views too, e.g.

* Alice receives quoted reply from Bob, starting downloading thumbnail.
* App is killed before download completes.
* Alice launches app, looks at conversation with Bob.  Expects quote-thumbnail download to retry (or _maybe_ tap to retry but I suspect we'll just want to retry automatically).

Reply:
I posted this comment in MessagesManager after opening the PR so maybe you missed it but I'll repost here:

> We don't currently handle retry.
> 
> Keep in mind, most quotes will be a recent message, which allows the thumbnail to be generated locally rather than fetched over the network, so I think it makes sense to prefer a simple rather than perfect solution here.
>
> With attachments the pattern is for a tap-to-retry. We could do something similar with thumbnail attachments. WDYT @charlesmchen?

Reply:
Yup just saw it.  Replied above.